// Copyright 2015 Google Inc.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
////////////////////////////////////////////////////////////////////////////////
syntax = "proto2";

package deepchem.models.tensorflow_models;

// Neural network model configuration, used mostly for
// de/serializing model parameters for a given execution from/to disk.
message ModelConfig {
  message Parameter {
    optional string name = 1;
    optional string description = 10;

    // See oneof user guide:
    // http://sites/protocol-buffers/user-docs/miscellaneous-howtos/oneof
    oneof value {
      float float_value = 2;
      int32 int_value = 3;
      string string_value = 4;
      bool bool_value = 5;
    }

    repeated float float_list = 6;
    repeated int32 int_list = 7;
    repeated string string_list = 8;
    repeated bool bool_list = 9;
  };

  repeated Parameter parameter = 1;
  optional string description = 2;
}
